src/PlusDataCollection/Atracsys/vtkPlusAtracsysTracker.cxx

```diff
@@ -0,0 +1,396 @@
+  float MaxMeanRegistrationErrorMm = 2.0;
+  int ActiveMarkerPairingTimeSec = 0;
+
+  // type of tracker connected
+  AtracsysTracker::DEVICE_TYPE DeviceType = AtracsysTracker::UNKNOWN_DEVICE;
+
+  // matches plus tool id to .ini geometry file names/paths
+  std::map<std::string, std::string> PlusIdMappedToGeometryFilename;
+
+  // matches fusionTrack internal tool geometry ID to Plus tool ID for updating tools
+  std::map<int, std::string> FtkGeometryIdMappedToToolId;
+
+  // Atracsys API wrapper class handle
+  AtracsysTracker* Tracker = new AtracsysTracker();
+};
+
+//----------------------------------------------------------------------------
+vtkPlusAtracsysTracker::vtkPlusAtracsysTracker()
+  : vtkPlusDevice()
+  , Internal(new vtkInternal(this))
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::vtkPlusAtracsysTracker()");
+
+  this->FrameNumber = 0;
+  this->StartThreadForInternalUpdates = true;
+  this->InternalUpdateRate = 60;
+}
+
+//----------------------------------------------------------------------------
+vtkPlusAtracsysTracker::~vtkPlusAtracsysTracker()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::~vtkPlusAtracsysTracker()");
+
+  delete Internal;
+  Internal = nullptr;
+}
+
+//----------------------------------------------------------------------------
+void vtkPlusAtracsysTracker::PrintSelf(ostream& os, vtkIndent indent)
+{
+  Superclass::PrintSelf(os, indent);
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::ReadConfiguration(vtkXMLDataElement* rootConfigElement)
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::ReadConfiguration");
+
+  XML_FIND_DEVICE_ELEMENT_REQUIRED_FOR_READING(deviceConfig, rootConfigElement);
+
+  XML_READ_SCALAR_ATTRIBUTE_NONMEMBER_OPTIONAL(int, MaxMissingFiducials, this->Internal->MaxMissingFiducials, deviceConfig);
+  if (this->Internal->MaxMissingFiducials < 0 || this->Internal->MaxMissingFiducials > 3)
+  {
+    LOG_WARNING("Invalid maximum number of missing fiducials provided. Must be between 0 and 3 inclusive. Maximum missing fiducials has been set to its default value of 0.");
+    this->Internal->MaxMissingFiducials = 0;
+  }
+
+  XML_READ_SCALAR_ATTRIBUTE_NONMEMBER_OPTIONAL(float, MaxMeanRegistrationErrorMm, this->Internal->MaxMeanRegistrationErrorMm, deviceConfig);
+  if (this->Internal->MaxMeanRegistrationErrorMm < 0.1 || this->Internal->MaxMeanRegistrationErrorMm > 5.0)
+  {
+    LOG_WARNING("Invalid maximum mean registration error provided. Must be between 0.1 and 5 mm inclusive. Maximum mean registration error has been set to its default value of 5.0mm.");
+    this->Internal->MaxMeanRegistrationErrorMm = 2.0;
+  }
+
+  XML_READ_SCALAR_ATTRIBUTE_NONMEMBER_OPTIONAL(int, ActiveMarkerPairingTimeSec, this->Internal->ActiveMarkerPairingTimeSec, deviceConfig);
+  if (this->Internal->ActiveMarkerPairingTimeSec > 60)
+  {
+    LOG_WARNING("Are you sure you want to leave active marker pairing enabled for more than 60 seconds?");
```

Comment:
This should not be a question - the user has no way of answer it. Rephrase it to a simple statement, something like: "Warning: active marker pairing time is set to ...sec. Tracking will not start until this time period is over."